packages/remix-serve/index.ts

```diff
@@ -1,30 +1,55 @@
 import express from "express";
+import type { Express } from "express";
 import compression from "compression";
 import morgan from "morgan";
 import { createRequestHandler } from "@remix-run/express";
+import type { RemixConfig } from "@remix-run/config";
+
+export function createApp(buildPathOrConfig: RemixConfig): Express;
+
+export function createApp(buildPathOrConfig: string): Express;
+
+export function createApp(
+  serverBuildPathOrConfig: string | RemixConfig,
+  mode = "production"
+): Express {
+  let serverBuildPath: string;
+  let assetsBuildDirectory: string;
+  let publicPath: string;
+  if (typeof serverBuildPathOrConfig === "string") {
+    // TODO: Deprecate using a string as 1st argument. Keep this behavior for
+    // now to avoid breaking changes.
+    serverBuildPath = serverBuildPathOrConfig;
+    assetsBuildDirectory = "public/build";
+    publicPath = "/build/";
+  } else {
+    serverBuildPath = serverBuildPathOrConfig.serverBuildPath;
+    assetsBuildDirectory = serverBuildPathOrConfig.assetsBuildDirectory;
+    publicPath = serverBuildPathOrConfig.publicPath;
+  }
```

Comment:
This could be simplified to

```suggestion
  let { assetsBuildDirectory, publicPath, serverBuildPath } = typeof serverBuildPathOrConfig === "string"
    // TODO: Deprecate using a string as 1st argument. Keep this behavior for
    // now to avoid breaking changes.
    ? { 
        assetsBuildDirectory: "public/build",
        publicPath: "/build/",
        serverBuildPath: serverBuildPathOrConfig,
      }
    : serverBuildPathOrConfig
```